### PR TITLE
ui: simplify onQueryChange callbacks

### PR DESCRIPTION
**Pull request by harunurhan**
Also fixes going back issue when a search link is clicked

### CODE REVIEW

File: ui/src/search/helper.js [hunk 1]
```diff
@@ -32,6 +42,35 @@ export default class SearchHelper {
     );
   }
 
+  updateLocation(urlSuffix = '') {
+    const searchUrl = `${this.getPathname()}?${this.getQueryString()}${urlSuffix}`;
+    if (this.isInitialQueryUpdate() || this.dueToNavigationToSearchPage) {
```
Comment (monaawi) [inline ui/src/search/helper.js:47]: So in the end we check for ```initialQueryUpdate``` even if it is embedded search? Like I said, it is worth it to be able to get rid of the code duplication. It looks better :+1:

File: ui/src/actions/search.js [hunk 2]
```diff
@@ -116,8 +120,18 @@ export function searchQueryUpdate(namespace, query) {
...
+      namespace,
+      prevState,
+      nextState,
+      dispatch,
+      dueToNavigationToSearchPage
+    );
+    searchConfig[namespace].onQueryChange(
+      helper,
+      dispatch,
+      dueToNavigationToSearchPage
```
Comment (monaawi) [inline ui/src/actions/search.js:133]: how come we didn't need this before for the synclocation? I didn't understand what change broke this

File: ui/src/search/helper.js (hunk 1 above)
Comment (harunurhan) [reply ui/src/search/helper.js:47]: no we don't.. embedded ones don't update location (yet)

File: ui/src/actions/search.js (hunk 2 above)
Comment (harunurhan) [reply ui/src/actions/search.js:133]: https://github.com/inspirehep/inspirehep/blob/e233cb503fdf9e205aaa2c43197b59c96a681b99/ui/src/actions/search.js#L53-L64

That one.

**Pull request merged**